lib/test-utils/src/Test/Hspec/Extra.hs

```diff
@@ -97,27 +101,32 @@ aroundAll acquire =
         release  <- newEmptyMVar
         done     <- newEmptyMVar
 
-        pid <- async $ do
-            acquire $ \a -> do
-                putMVar resource a
-                await release
-            unlock done
+        pid <- async $ finally (unlock done) $ acquire $ \a -> do
+            putMVar resource a
+            await release
 
-        race (wait pid) (takeMVar resource) >>= \case
-            Left _ ->
-                throwIO $ userError "aroundAll: failed to setup"
-            Right a -> pure $ (a,) $ do
+        let cleanup = do
                 unlock release
                 await done
 
+        race (wait pid) (takeMVar resource) >>= \case
+            Left _ -> throwString "aroundAll: failed to setup"
+            Right a -> pure (a, cleanup)
+
+    await :: MVar () -> IO ()
+    await = takeMVar
+
+    unlock  :: MVar () -> IO ()
+    unlock = flip putMVar ()
+
 -- | A drop-in replacement for 'it' that'll automatically retry a scenario once
 -- if it fails, to cope with potentially flaky tests.
 --
 -- It also has a timeout of 10 minutes.
 it :: HasCallStack => String -> ActionWith ctx -> SpecWith ctx
-it = itWithCustomTimeout (10*minute)
+it = itWithCustomTimeout (60*minutes)
   where
-    minute = 60
+    minutes = 10
```

Comment:
I get this change, but this confuses me. Doesn't matter, but it's interesting to see how brains pick those things up.

Reply:
I needed to temporarily reduce the timeout to 1 minute - so in this form you can just edit the minutes variable to set the timeout.